Jenkins integration

brianhy opened the pull request:
Preliminary Jenkins integration

Jenkins will need this checked in before it'll actually run, so we need to check-in, then run, then debug any issues.

brianhy committed:
Add Jenkins support and csdisco-specific readme

No changes to the project itself yet.

brianhy committed:
Remove net46 framework from net-core

When this is present the build will break on the mac b/c it doesn't have .net 4.6 installed.  Instead of trying to work around that (which isn't needed since we want netstandard for .net core support), we instead deprecate it.

We didn't go so far as to remove the .Net framework dirs as that would result in lots of git merge issues if we take new drops from master.

brianhy committed:
Update nuspec to release a "csdisco" version of the library

brianhy committed:
For now set major version to 0 while we work kinks out of jenkins.

brianhy committed:
Temporarily support releasing from the jenkins-integration branch

brianhy committed:
Force publishing from the PR

In Jenkins, a PR isn't run from the branch, so the releaseBranch check wasn't doing anything.  Instead, we'll just always run the publish to test publish to artifactory.

discoci2 commented:
:rocket: This PR has been checked for open source licensing and security compliance.

Analysis Results: 
```
 Test Passed! 0 issues found
 
```

For more information, see:
* [Open Source Licensing and Security at DISCO](https://csdisco.atlassian.net/wiki/spaces/DO/pages/1079150488/Open+Source+Licensing+and+Security+at+DISCO)
* [See ical.net in Fossa](https://csdisco.fossa.app/)
* [Jenkins job that performed this check](http://jenkins.casilabs.net:8080/job/open-source-license-scanner/4270/)

brianhy committed:
Override version to be 0.0.1

Nuget publish succeeded, but I don't see it in nuget.  The log output makes me wonder if the build # is problematic:
> 12:01:45 Build successfully deployed. Browse it in Artifactory under https://csdisco.jfrog.io/csdisco/webapp/builds/csdisco-ical-net%20::%20PR-1/3

brianhy committed:
Fix version overwriting and make sure configuration is Release

brianhy committed:
Put nupkg into a different directory

brianhy committed:
Fix output dir (missing 's')

brianhy committed:
Capitalize csdisco

brianhy committed:
Have jenkins build populate the build description

Now that build description will include build version, we need to feed that to the build.ps1 to keep those in sync.
